Exercise from Think Stats, 2nd Edition (thinkstats2.com)<br>
Allen Downey

In [1]:
import nsfg
df = nsfg.ReadFemPreg()
df

Print value counts for <tt>birthord</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611933)

df.birthord.value_counts().sort_index()

Yes, everything seems to be alligned with the link:

value	label	 	Total
.	INAPPLICABLE	 	4445
1	1ST BIRTH	 	4413
2	2ND BIRTH	 	2874
3	3RD BIRTH	 	1234
4	4TH BIRTH	 	421
5	5TH BIRTH	 	126
6	6TH BIRTH	 	50
7	7TH BIRTH	 	20
8	8TH BIRTH	 	7
9	9TH BIRTH	 	2
10	10TH BIRTH	 	1
 	Total	 	13593

Print value counts for <tt>prglngth</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611931)

In [27]:
import numpy as np
baby_length_grouped = np.array([])
baby_length_grouped = np.append(baby_length_grouped, )
np.append(baby_length_grouped, 2)

array([ 1.,  2.])

In [33]:
import numpy as np
import pandas as pd

baby_length_grouped = np.array([])

for baby_length in df.prglngth:
    if baby_length < 14:
        baby_length_grouped = np.append(baby_length_grouped,"small")
    elif baby_length >= 14 and baby_length <27:
        baby_length_grouped = np.append(baby_length_grouped,"medium")
    else:
        baby_length_grouped = np.append(baby_length_grouped,"large")

s = pd.Series(baby_length_grouped)
print s.value_counts()

large     9278
small     3522
medium     793
dtype: int64


Print value counts for <tt>agepreg</tt> and compare to results published in the [codebook](http://www.icpsr.umich.edu/nsfg6/Controller?displayPage=labelDetails&fileCode=PREG&section=A&subSec=8016&srtLabel=611935).

Looking at this data, please remember my comments in the book about the obligation to approach data with consideration for the context and respect for the respondents.

Yes - the baby's lengths seem to allign:

value	label	 	            Total
0-13	13 WEEKS OR LESS	 	3522
14-26	14-26 WEEKS	 	        793
27-50	27 WEEKS OR LONGER	 	9278
 	Total

Compute the mean birthweight.

In [34]:
df.totalwgt_lb.mean()

7.265628457623368

Create a new column named <tt>totalwgt_kg</tt> that contains birth weight in kilograms.  Compute its mean.  Remember that when you create a new column, you have to use dictionary syntax, not dot notation.

Look through the codebook and find a variable, other than the ones mentioned in the book, that you find interesting.  Compute values counts, means, or other statistics.

Create a boolean Series.

In [4]:
df.outcome == 1

0      True
1      True
2      True
3      True
4      True
5      True
6      True
7      True
8      True
9      True
10     True
11     True
12     True
13    False
14    False
...
13578     True
13579     True
13580    False
13581     True
13582    False
13583    False
13584     True
13585    False
13586    False
13587    False
13588     True
13589    False
13590    False
13591     True
13592     True
Name: outcome, Length: 13593, dtype: bool

Use a boolean Series to select the records for the pregnancies that ended in live birth.

In [5]:
live = df[df.outcome == 1]
len(live)

9148

Count the number of live births with <tt>birthwgt_lb</tt> between 0 and 5 pounds (including both).  The result should be 1125.

In [6]:
len(live[(live.birthwgt_lb >= 0) & (live.birthwgt_lb <= 5)])

1125

Count the number of live births with <tt>birthwgt_lb</tt> between 9 and 95 pounds (including both).  The result should be 798 

Use <tt>birthord</tt> to select the records for first babies and others.  How many are there of each?

In [7]:
firsts = df[df.birthord==1]
others = df[df.birthord>1]
len(firsts), len(others)

(4413, 4735)

Compute the mean weight for first babies and others.

In [8]:
firsts.totalwgt_lb.mean()

7.201094430437772

In [9]:
others.totalwgt_lb.mean()

7.3258556149732623

Compute the mean <tt>prglngth</tt> for first babies and others.  Compute the difference in means, expressed in hours.